src/storage.rs

```diff
@@ -297,4 +326,12 @@ mod tests {
         assert!(eq_bytecodes(&bytecode, &evm_code));
         assert_eq!(bytecode, Bytecode::from(evm_code));
     }
+
+    #[test]
+    fn eof_bytecodes() {
+        let bytecode = Bytecode::Eof(Arc::new(Eof::decode(EOF_BYTECODE).unwrap()));
+        let evm_code = EvmCode::from(bytecode.clone());
+        assert!(eq_bytecodes(&bytecode, &evm_code));
+        assert_eq!(EOF_BYTECODE, Bytecode::from(evm_code).original_bytes());
```

Comment:
Perhaps `assert_eq!(bytecode, Bytecode::from(evm_code));` is tighter as it enforces us to convert back to `Bytecode::Eof` instead of something else that happens to has the same original bytes?